### DISCUSSION TITLE
Serverless uploading to S3?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#4117] [Q&A]
When PocketBase is using S3 for file storage, are uploads performed via signed URL directly to S3, or do they need to first be uploaded to pb and then transferred to S3?

https://aws.amazon.com/blogs/compute/uploading-to-amazon-s3-directly-from-a-web-or-mobile-application/

From our discussion https://discord.com/channels/1128192380500193370/1193826604783247420/1194200789275639830

**Comment by ganigeorgiev** [comment]
I'm not able to access the above discord link (or at least it doesn't seem to load anything for me), but all S3 operations are performed over the S3 APIs with the configured credentials (eg. `PutObject`).

When using S3, uploaded files are not stored on the PocketBase server, although it is possible, depending on the file size, the file to be kept in memory or in a temp location during the processing.

**Reply by benallfree** [reply]
Sorry, apparently you need to be invited to the server first https://discord.gg/nVTxCMEcGT

Anyway, some users are hitting CloudFlare's 100MB request body limit and are looking for ways around it.

**Reply by ganigeorgiev** [reply]
I see. Unfortunately other than manually splitting the file and upload it on chunks I don't know if this can be avoided (_it was suggested recently in https://github.com/pocketbase/pocketbase/issues/3691 but since it requires completely different upload handling, both for the client-side sdk and server-side, it is left for later consideration_).

**Comment by jonshipman** [comment]
If you're just looking for a solution, setting the field in Pocketbase to "Text" and interfacing with S3 directly in the client. Then saving the S3 document location as a text value in Pocketbase.

Not ideal, but you can just create a function in the client to do both actions at once.
